### PR TITLE
samples: indexing samples conform to rubric

### PR DESCRIPTION
**Pull request by laljikanjareeya**
This PR Contains Indexing Samples that covers the new Canonical format.
It does not remove the old samples.
Towards googleapis/google-cloud-node#7377

### CODE REVIEW

**Review by AVaksman — APPROVED** [review]

File: samples/index-create-storing.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+      console.log('Added the AlbumsByAlbumTitle2 index.');
+    } catch (err) {
+      console.error('ERROR:', err);
+    } finally {
+      // Close the database when finished.
+      database.close();
+    }
+  }
+  createStoringIndex().catch(console.error);
```
Comment (skuruppu) [inline samples/index-create-storing.js:66]: The examples given don't do this `catch()` call. I would like to understand why it's here.

File: samples/index-read-data-with-storing.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  // Imports the Google Cloud Spanner client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  // Instantiates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function readDataWithStoringIndex() {
```
Comment (skuruppu) [inline samples/index-read-data-with-storing.js:42]: Should we add the comment from the original sample?
```
  // "Storing" indexes store copies of the columns they index
  // This speeds up queries, but takes more space compared to normal indexes
  // See the link below for more information:
  // https://cloud.google.com/spanner/docs/secondary-indexes#storing_clause
```

File: samples/system-test/spanner.test.js [hunk 3]
```diff
@@ -440,11 +439,7 @@ describe('Spanner', () => {
 
   it('should respect query boundaries when querying an example table with an index', async () => {
     const output = execSync(
-      `${indexingCmd} queryIndex ${INSTANCE_ID} ${DATABASE_ID} ${PROJECT_ID} -s Ardvark -e Zoo`
-    );
-    assert.match(
-      output,
-      /AlbumId: 1, AlbumTitle: Total Junk, MarketingBudget:/
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:447]: Why is this check removed?

**Review by skuruppu — COMMENTED** [review]
Sorry for the huge delay in reviewing this. The change looks good. I have a couple of questions in the comments.

File: samples/index-create-storing.js (hunk 1 above)
Comment (laljikanjareeya) [reply samples/index-create-storing.js]: No, we don't need the `catch` block as we are handing `unhandledRejection` on the process.
Removed!

File: samples/index-read-data-with-storing.js (hunk 2 above)
Comment (laljikanjareeya) [reply samples/index-read-data-with-storing.js:46]: Added!

File: samples/system-test/spanner.test.js (hunk 3 above)
Comment (laljikanjareeya) [reply samples/system-test/spanner.test.js]: Reverted.

**Review by skuruppu — APPROVED** [review]

**Comment by laljikanjareeya** [comment]
@skuruppu any idea on `OwlBot Post Processor` Build step. it's taking longer than expected.

**Comment by skuruppu** [comment]
@laljikanjareeya would you be able check the docs failure?

**Comment by laljikanjareeya** [comment]
@skuruppu `docs` test failing because the newly added sample in this PR will not be available in the master branch while docs test checking link on a master branch instead of on the same branch.
Ref:
image

**Comment by skuruppu** [comment]
Ah yeah that problem. I'm following up with the Node team to see how this can be fixed. I'll follow up.

**Pull request merged**